### DISCUSSION TITLE
Engine in Rust

### DISCUSSION THREAD

**Discussion by ecatugy** [postcss/postcss#1942] [Ideas]
The new version of TailwindCSS has switched to Lightning CSS for performance reasons since the engine is written in Rust. Any plans to do the same for PostCSS? I believe if this isn’t considered, we’ll see major players migrating to Lightning CSS in the long run.

**Comment by ai** [comment]
I am pretty happy with migration to Lightning CSS.

Big open source project maintenance is a slavery job. This community migration to Lightning CSS mean that I will be finally free 🥹.

So, no, I don't want to rewrite the project to Rust. It is now Lightning CSS team job 😅. Also, their implementation is close to ideal, there is nothing I can do better.

**Reply by romainmenke** [reply]
I do hope that you will be free from PostCSS maintenance in the near future @ai, but I don't think LightningCSS is the thing that will make this possible.

I don't think LightningCSS is the successor to PostCSS because:
- it is written in Rust and contributions are too hard for most developers who focus on CSS
- it is fast, but PostCSS is also very fast and the difference is not large enough for most users
- LightningCSS has too many open bugs that take too long to get fixed (see point 1)
- LightningCSS is lossy, it is impossible to build dev tooling on top of it (things like Stylelint)
- LightningCSS is not forward compatible, new CSS breaks your build until LightningCSS is updated to support the new features.

I was hoping it would be a good replacement, because then I would also be free from PostCSS Preset Env. Don't take this as a critique on LightningCSS, it isn't :)

I believe there is still enormous value in PostCSS especially for things like Stylelint. These are impossible and will remain impossible in LightningCSS.

But PostCSS is also largely feature complete, so maybe you can be free from PostCSS by declaring it "finished"? :) 

If there is anything I can do to help, please let me know!
